Makefile

```diff
@@ -0,0 +1,40 @@
+#! /usr/bin/env make -f
+# This Makefile assumes an activated Python virtual environment.
+
+# Make sure that Python's virtual environment is activated, and
+# all tools are available.
+ifndef VIRTUAL_ENV
+  $(error Please activate the Python virtual environment)
+endif
+
+.PHONY:	all
+all:	dist
```

Comment:
I'm not sure if only running `dist` here makes sense. When I run `make all` I expect to run all the stages.

We could change `dist` to `build` and then add all the following stages to `all`?

Reply:
At least GNU Make defines a set of [Standard Targets](https://www.gnu.org/software/make/manual/html_node/Standard-Targets.html)<sup>*</sup> and it defines the “dummy target” `all` as

> Compile the entire program. This should be the default target. This target need not rebuild any documentation files; […]

Considering that this is a Python package where we don’t actually compile anything, I used the `dist` target as the closest sensible target for the `all` dummy:

> Create a distribution tar file for this program. The tar file should be set up so that the file names in the tar file start with a subdirectory name which is the name of the package it is a distribution for. This name can include the version number.

> @behnazh When I run `make all` I expect to run all the stages.

With “stages” do you mean build targets? If the package has been built already, then what would you expect `make all` to do?

—————
<sup>*</sup> See also [this Stackoverflow question](https://stackoverflow.com/questions/2514903/what-does-all-stand-for-in-a-makefile).

Reply:
As you mentioned because Python packages don't compile anything, it might not make sense to strictly follow the conventions in GNU Make.

I expect `make all` to run the `quick-check`, `check`, `test`, and if the package has not already been built `dist` (`build`) targets.

Reply:
Commit https://github.com/jenstroeger/python-package-template/pull/74/commits/68d6a919002d5c9f327dc19f57fbd6a97bbae32c.